### PR TITLE
Only depend on tier in APIConfig which represents apiTier

### PR DESCRIPTION
**Pull request by suksw**
### Purpose
Remove apitier field from AuthenticationContext and APIKeyValidationInfoDTO and only depend on tier field in APIConfig.

### Issues
Fixes https://github.com/wso2/product-microgateway/issues/2264

### Automation tests
 - Unit tests added: Yes/No
 - Integration tests added: Yes/No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/util/FilterUtils.java [hunk 1]
```diff
@@ -516,26 +513,6 @@ public static String buildUsernameWithTenant(String username, String tenantDomai
...
-
-        if (!apiTier.isEmpty() && !ThrottleConstants.UNLIMITED_TIER.equalsIgnoreCase(apiTier)) {
-            return apiTier;
-        }
-        if (!resourceTier.isBlank()) {
-            return resourceTier;
-        }
-        return ThrottleConstants.UNLIMITED_TIER;
-    }
-
```
Comment (suksw) [inline enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/util/FilterUtils.java:538]: This method becomes unused with this PR. The scenario is already handled in ThrottleFilter.

File: enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/websocket/WebSocketMetaDataFilter.java [hunk 2]
```diff
@@ -74,8 +74,6 @@ public void init(APIConfig apiConfig, Map<String, String> configProperties) {
                     getNullableStringValue(authenticationContext.getApplicationTier()));
             requestContext.addMetadataToMap(MetadataConstants.TIER,
                     getNullableStringValue(authenticationContext.getTier()));
-            requestContext.addMetadataToMap(MetadataConstants.API_TIER,
-                    getNullableStringValue(authenticationContext.getApiTier()));
```
Comment (VirajSalaka) [inline enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/websocket/WebSocketMetaDataFilter.java:78]: We need to populate metadata somehow. Hence we need to populate it using the APIConfig.

Comment (suksw) [reply enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/websocket/WebSocketMetaDataFilter.java:78]: Fixed with https://github.com/wso2/product-microgateway/pull/2496/commits/d004fc72875effcfcd566a555a9f2c2117b67d14

**Review by praminda — APPROVED** [review]

**Review by VirajSalaka — APPROVED** [review]

**Comment by choreo-cicd** [comment]
[failed] dev-deployment : 20211215.32

**Comment by choreo-cicd** [comment]
[succeeded] dev-deployment : 20211215.43

**Pull request merged**